app/Http/Controllers/PostController.php

```diff
@@ -103,7 +103,20 @@ public function update(Request $request, string $id): void
      * Remove the specified resource from storage.
      * @param string $id
      */
-    public function destroy(string $id): void
+    public function destroy(string $id): RedirectResponse
     {
+        //$idが存在するか/削除権限があるかバリデーション
+        if (!Post::idExists($id)){
+            abort(404);
+        }
+        $post = Post::find($id);
```

Comment:
```suggestion
        $post = Post::find($id);
        if ($post === null){
            abort(404);
        }
```
でいいんですよね
```suggestion
        $post = Post::findOrFail($id);
```
っていうやり方もあります。
三宅さんのやり方だと、存在チェックに1クエリ、取得に1クエリで合計2クエリ使うわけですが、1回でいいよねということですね

Reply:
ありがとうございます。これからはクエリの回数を意識してみます。コミット「[存在チェックを2クエリから1クエリに変更](https://github.com/miya1031/BBS-laravel-/pull/20/commits/dfe22a63db251f35e290b02943986775d968d54e)」にて変更しました。